src/ngraph/op/cum_sum.cpp

```diff
@@ -0,0 +1,74 @@
+//*****************************************************************************
+// Copyright 2017-2019 Intel Corporation
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+//*****************************************************************************
+
+#include "ngraph/op/cum_sum.hpp"
+#include "ngraph/graph_util.hpp"
+#include "ngraph/op/broadcast.hpp"
+#include "ngraph/op/constant.hpp"
+
+using namespace std;
+using namespace ngraph;
+
+constexpr NodeTypeInfo op::CumSum::type_info;
+
+op::CumSum::CumSum(const Output<Node>& arg,
+                   const Output<Node>& axis,
+                   const int exclusive,
+                   const int reverse)
+    : Op({arg, axis})
+    , m_exclusive(exclusive)
+    , m_reverse(reverse)
+{
+    auto const_op = as_type_ptr<op::Constant>(axis.get_node_shared_ptr());
+    m_axis = *(static_cast<const int64_t*>(const_op->get_data_ptr()));
```

Comment:
What will happen if `axis` isn't `Constant`?
Will we get a segfault here?

Reply:
Output from debugging ngraph-onnx tests:
```
tests/test_backend.py::OnnxBackendNodeModelTest::test_cumsum_1d_cpu
Thread 1 "python" received signal SIGSEGV, Segmentation fault.
```
``` 
#4  0x00007fff4f7c3509 in ngraph::op::CumSum::CumSum (this=0x157e910, arg=..., axis=..., exclusive=0, reverse=0)
    at /root/projects/ngraph/src/ngraph/op/cum_sum.cpp:36
36          m_axis = *(static_cast<const int64_t*>(const_op->get_data_ptr()));
```